src/include/miopen/tensor_layout.hpp

```diff
@@ -27,14 +27,24 @@
 #define GUARD_TENSOR_LAYOUT_HPP
 
 #include <map>
+#include <unordered_map>
 #include <algorithm>
 #include <vector>
+#include <numeric>
 #include <string>
 #include <iterator>
 #include <miopen/errors.hpp>
 
```

Comment:
TODO: wrap this header into a proper class, not a collection of static functions.